### ISSUE TITLE
Design Concerns

### ISSUE THREAD

**Issue by alfphaderp** [quickTwix898/login-experimental#4]
This whole thing you're doing with having the database file open with a reader, writer, and keeping track of positions is tedious and things can easily go wrong. If you're just practicing file io, by all means, go ahead and just close this issue, but a more practical and easy alternative would be to just store your information in a Map data structure and serialize/deserialize it.

**Comment by alfphaderp** [comment]
Furthermore, it looks like to verify a username is in the database, you're scanning through line by line to find the user. This is extremely inefficient as it would take linear time to find users. Using a HashMap could speed things up significantly by allowing you to perform lookups in constant time.

**Comment by quickTwix898** [comment]
I agree. The main focus of the project was using security/cryptography libraries in Java. If this were to be actually implemented in another project, the whole text file idea would be thrown out due to scalibility concerns. Thanks
